### PR TITLE
FIX: fix the post request function

### PR DESCRIPTION
**Pull request by asamolion**
Steps to verify:

1. Add the necessary configuration in private.py to enable the external API call.
2. Create a server that can receive that request. You can use express.js or Flask. Any micro-framework will do.
3. Run an event and see if it is received on the server side.

### CODE REVIEW

File: openedx_caliper_tracking/processor.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def deliver_caliper_event(caliperized_event, event_type):
             headers={
                 'Authorization': 'Bearer {}'.format(
                     settings.CALIPER_DELIVERY_AUTH_TOKEN),
-                'Content-Type': 'application/json',
+                'Content-Type': 'application/vnd.kafka.json.v2+json',
             },
             json={
```
Comment (imhassantariq) [inline openedx_caliper_tracking/processor.py]: According to requests documentation, if we use `json` variable in requests call `Content-Type` in the header will set to `application/json` but we need `application/vnd.kafka.json.v2+json` please look into this.

REF: [Link](http://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests)

Comment (HamzaIbnFarooq) [reply openedx_caliper_tracking/processor.py]: good catch @imhassantariq

Comment (asamolion) [reply openedx_caliper_tracking/processor.py]: I checked it with `json` and it worked fine. However, I still modified and replaced with data. Just being careful.

**Review by mumarkhan999 — APPROVED** [review]

File: openedx_caliper_tracking/loggers.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def get_caliper_logger(logger_name):
 
     logger = logging.getLogger(logger_name)
 
-    if settings.DEBUG:
+    # checks if the environment is either devstack or test
```
Comment (danialmalik) [inline openedx_caliper_tracking/loggers.py:50]: Do we need this comment here?

Comment (asamolion) [reply openedx_caliper_tracking/loggers.py:50]: Yes, because the check is very obscure.

File: openedx_caliper_tracking/tests/tests.py [hunk 3]
```diff
@@ -6,10 +6,12 @@
 import json
 import os
 
-from django.test import TestCase
+from django.conf import settings
+from django.test import TestCase, override_settings, modify_settings
```
Comment (danialmalik) [inline openedx_caliper_tracking/tests/tests.py]: `modify_settings` is not used anywhere so it should be removed.

Comment (HamzaIbnFarooq) [reply openedx_caliper_tracking/tests/tests.py]: @asamolion

Comment (asamolion) [reply openedx_caliper_tracking/tests/tests.py]: @HamzaIbnFarooq Already fixed this. Check below this comment.

**Review by danialmalik — APPROVED** [review]

**Comment by HamzaIbnFarooq** [comment]
Manual QA done,
`Flask server` wasn't able to decode sent `JSON` data due to `content-type= 'application/vnd.kafka.json.v2+json'` but when sent on kafka server it gets a 200 response(as QA done on dev server by @asamolion) . Other than that everything is working fine. @asamolion you can update the documentation and publish it on PyPI.

**Comment by asamolion** [comment]
Thanks @HamzaIbnFarooq

**Pull request merged**